src/main/lombok/org/heigit/ohsome/ohsomeapi/inputprocessing/InputProcessor.java

```diff
@@ -391,28 +392,44 @@ public boolean compareKeysValues(String[] keys, String[] keys2, String[] values,
   }
 
   /**
-   * Processes the properties parameter used in data-extraction ressources and sets the respective
-   * boolean values includeTags, includeOSMMetadata and unclippedGeometries.
+   * Processes the properties parameter used in data-extraction resources and sets the respective
+   * boolean values includeTags, includeOSMMetadata and unclippedGeometries. It sets also the
+   * boolean value includeContributionTypes for the contribution endpoints.
    * 
    * @throws BadRequestException if the properties parameter contains invalid values
    */
   public void processPropertiesParam() {
     String[] properties =
         splitParamOnComma(createEmptyArrayIfNull(requestParameters.get("properties")));
-    if (properties.length > 3) {
-      throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM);
-    }
-    for (String property : properties) {
-      @Deprecated
-      boolean oldUnclippedParameter = "unclipped".equalsIgnoreCase(property);
-      if ("tags".equalsIgnoreCase(property)) {
-        this.includeTags = true;
-      } else if ("metadata".equalsIgnoreCase(property)) {
-        this.includeOSMMetadata = true;
-      } else if (oldUnclippedParameter) {
-        this.clipGeometry = false;
-      } else {
-        throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM);
+    if (!RequestUtils.isContributionsExtraction(requestUrl)) {
+      for (String property : properties) {
+        @Deprecated
+        boolean oldUnclippedParameter = "unclipped".equalsIgnoreCase(property);
+        if ("tags".equalsIgnoreCase(property)) {
+          this.includeTags = true;
+        } else if ("metadata".equalsIgnoreCase(property)) {
+          this.includeOSMMetadata = true;
+        } else if (oldUnclippedParameter) {
+          this.clipGeometry = false;
+        } else {
+          throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM);
+        }
+      }
+    } else {
+      for (String property : properties) {
+        @Deprecated
+        boolean oldUnclippedParameter = "unclipped".equalsIgnoreCase(property);
+        if ("tags".equalsIgnoreCase(property)) {
+          this.includeTags = true;
+        } else if ("metadata".equalsIgnoreCase(property)) {
+          this.includeOSMMetadata = true;
+        } else if (oldUnclippedParameter) {
+          this.clipGeometry = false;
+        } else if ("contributionTypes".equalsIgnoreCase(property)) {
+          this.includeContributionTypes = true;
+        } else {
+          throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM_CONTR);
+        }
       }
     }
   }
```

Comment:
I think you can write this method in an easier way with just one for-each loop. Please take a look at my suggested approach and check, but I think it should also work like that. At least the local tests worked fine.

```java
public void processPropertiesParam() {
    String[] properties =
        splitParamOnComma(createEmptyArrayIfNull(requestParameters.get("properties")));
    for (String property : properties) {
      @Deprecated
      boolean oldUnclippedParameter = "unclipped".equalsIgnoreCase(property);
      if ("tags".equalsIgnoreCase(property)) {
        this.includeTags = true;
      } else if ("metadata".equalsIgnoreCase(property)) {
        this.includeOSMMetadata = true;
      } else if (oldUnclippedParameter) {
        this.clipGeometry = false;
      } else if (RequestUtils.isContributionsExtraction(requestUrl)) {
        if ("contributionTypes".equalsIgnoreCase(property)) {
          this.includeContributionTypes = true;
        } else {
          throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM_CONTR);
        }
      } else {
        throw new BadRequestException(ExceptionMessages.PROPERTIES_PARAM);
      }
    }
  }
```